src/main/java/io/imply/druid/example/indexer/ExampleByteBufferInputRowParser.java

```diff
@@ -0,0 +1,113 @@
+package io.imply.druid.example.indexer;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.data.input.ByteBufferInputRowParser;
+import org.apache.druid.data.input.InputRow;
+import org.apache.druid.data.input.impl.MapInputRowParser;
+import org.apache.druid.data.input.impl.ParseSpec;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.parsers.Parser;
+
+import java.nio.ByteBuffer;
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+
+public class ExampleByteBufferInputRowParser implements ByteBufferInputRowParser
+{
+  public static final String TYPE_NAME = "exampleParser";
+
+  private final ParseSpec parseSpec;
+  private Parser<String, Object> parser;
+  private MapInputRowParser mapParser;
+
+  private final Base64.Decoder base64Decoder = Base64.getDecoder();
+
+  @JsonCreator
+  public ExampleByteBufferInputRowParser(
+      @JsonProperty("parseSpec") ParseSpec parseSpec
+  )
+  {
+    this.parseSpec = parseSpec;
+  }
+
+  @JsonProperty
+  @Override
+  public ParseSpec getParseSpec()
+  {
+    return parseSpec;
+  }
+
+  @Override
+  public ByteBufferInputRowParser withParseSpec(ParseSpec parseSpec)
+  {
+    return new ExampleByteBufferInputRowParser(parseSpec);
+  }
+
+  @Override
+  public List<InputRow> parseBatch(ByteBuffer input)
+  {
+    if (parser == null) {
+      parser = parseSpec.makeParser();
+      mapParser = new MapInputRowParser(parseSpec);
```

Comment:
Might make more sense to use a StringInputRowParser and let it do the parseBatching.